hack/update-licenses.sh

```diff
@@ -0,0 +1,186 @@
+# limitations under the License.
+
+# Generates a table of the licenses of the cluster-registry and its vendored
+# dependencies, and outputs it to a LICENSES file.
+#
+# Usage:
+#    $0 [--verify-only]
+
+set -euo pipefail
+
+export LANG=C
+export LC_ALL=C
+
+###############################################################################
+# Process package content
+#
+# @param package  The incoming package name
+# @param type     The type of content (LICENSE or COPYRIGHT)
+#
+process_content () {
+  local package=$1
+  local type=$2
+
+  local package_root
+  local ensure_pattern
+  local dir_root
+  local find_maxdepth
+  local find_names
+  local -a local_files=()
+
+  # Necessary to expand {}
+  case ${type} in
+      LICENSE) find_names=(-iname 'licen[sc]e*')
+               find_maxdepth=1
+               # Sadly inconsistent in the wild, but mostly license files
+               # containing copyrights, but no readme/notice files containing
+               # licenses (except to "see license file")
+               ensure_pattern="license|copyright"
+               ;;
+    # We search READMEs for copyrights and this includes notice files as well
+    # Look in as many places as we find files matching
+    COPYRIGHT) find_names=(-iname 'notice*' -o -iname 'readme*')
+               find_maxdepth=3
+               ensure_pattern="copyright"
+               ;;
+  esac
+
+  # Start search at package root
+  case ${package} in
+    github.com/*|golang.org/*|bitbucket.org/*)
+     package_root=$(echo ${package} |awk -F/ '{ print $1"/"$2"/"$3 }')
+     ;;
+    go4.org/*)
+     package_root=$(echo ${package} |awk -F/ '{ print $1 }')
+     ;;
+    *)
+     package_root=$(echo ${package} |awk -F/ '{ print $1"/"$2 }')
+     ;;
+  esac
+
+  # Find files - only root and package level
+  local_files=($(
+    for dir_root in ${package} ${package_root}; do
+      [[ -d ${DEPS_DIR}/${dir_root} ]] || continue
+
+      # One (set) of these is fine
+      find ${DEPS_DIR}/${dir_root} \
+          -xdev -follow -maxdepth ${find_maxdepth} \
+          -type f "${find_names[@]}"
+    done | sort -u))
+
+  local index
+  local f
+  index="${package}-${type}"
+  if [[ -z "${CONTENT[${index}]-}" ]]; then
+    for f in ${local_files[@]-}; do
+      # Find some copyright info in any file and break
+      if egrep -i -wq "${ensure_pattern}" "${f}"; then
+        CONTENT[${index}]="${f}"
+        break
+      fi
+    done
+  fi
+}
+
+
+#############################################################################
+# MAIN
+#############################################################################
+
+# Check bash version
+if ((${BASH_VERSINFO[0]}<4)); then
+  echo
+  echo "ERROR: Bash v4+ required."
+  # Extra help for OSX
+  if [[ "$(uname -s)" == "Darwin" ]]; then
+    echo
+    echo "Ensure you are up to date on the following packages:"
+    echo "$ brew install md5sha1sum bash"
+  fi
+  echo
+fi
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
```

Comment:
Probably works just as well, but perhaps we should explicitly dereference the zeroth index for clarity.

Reply:
That seems reasonable. Done.